source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,369 @@
+
+
+
+Clone the Linux kernel package 
+==============================
+Clone the existing kernel package repository from |CL| as a starting point.
+
+#. Clone the Linux kernel package from |CL|.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+      make clone_linux
+
+
+#. Navigate into the cloned package directory.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux/packages/linux
+
+.. note::
+    The "linux" package is the kernel that comes with |CL| in the kernel-native bundle.
+    You can alternatively use a different kernel variant as the base for modification. 
+    For a list of kernel package names which you can clone instead, see the `clearlinux-pkgs GitHub`_.
+
+
+
+Change the kernel version 
+=========================
+
+|CL| tends to use the latest kernel available from `kernel.org`_, the Linux 
+upstream. The kernel version that will be built can be changed in the 
+RPM SPEC file.
+
+While most packages in Clear Linux are typically packaged using 
+`autospec`_, the kernel is not. This means control files provided 
+by autospec are not available and changes must be made manually.
+
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+
+#. Modify the Version, Release, and Source0 URL entries at the top of the 
+   file to change the version of Linux kernel that will be compiled.
+
+   A list of current and available kernel release can be found on 
+   `kernel.org`_.
+
+   .. code-block:: bash
+      
+      Name:           linux
+      Version:        4.20.5
+      Release:        690
+      License:        GPL-2.0
+      Summary:        The Linux kernel
+      Url:            http://www.kernel.org/
+      Group:          kernel
+      Source0:        https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.20.5.tar.xz
+      Source1:        config
+      Source2:        cmdline
+
+   .. note::
+      Consider changing the Name in the RPM spec file to easily identify a modified kernel.
+
+#. Commit and save the changes to the file.
+
+
+
+Modify kernel configuration 
+===========================
+
+Existing kernel features and in-tree kernel modules can be enabled or 
+disabled in the kernel configuration file, :file:`.config` , at compile time.
+
+To manage unique changes that have been made to the kernel config file, 
+|CL| uses a kernel config fragment file named :file:`config-fragment`. 
+Managing kernel configuration changes with a configuration fragment file 
+instead of directly editing the :file:`.config` file helps identify the 
+unique configuration changes that have been made and makes applying any 
+future default configuration values easier. 
+
+The :file:`config-fragment` is the **only** file that is modified and is 
+eventually merged with the main :file:`.config`.
+
+
+
+#. Open the kernel :file:`config-fragment` file in an editor.
+   
+   Due to how |CL| packaging tools make use of :command:`mock` environments 
```

Comment:
They can use menuconfig if they work outside of mock, that is, directly in ```/var/lib/mock/clear-linux/root/builddir/build/BUILD/kernel-*```

Reply:
That makes sense technically. Is entering into the mock environment a flow that we want to support/encourage?